client/src/common/components/Header.tsx

```diff
@@ -1,161 +1,86 @@
+		return isLogin ? (
+			<div className="flex items-center gap-3 relative">
+				<button className="border-2 border-solid	border-primary_bg_color rounded-full px-3 text-sm">
+					Log out
+				</button>
+				<Hamburger
+					toggled={isOpen}
+					toggle={setOpen}
+					size={20}
+					distance="lg"
+					rounded
+				/>
 			</div>
-			{/* {!hasValidToken ? ( */}
-			{/* <div className="text-base 
-								box-border 
-								border-2 
-								w-20 
-								rounded-full 
-								cursor-pointer 
-								font-base 
-								hover:bg-primary_bg_color 
-								mr-5 
-								text-center">
-					<Link to="/login">Login</Link>
-				</div>
-				<div className="bg-primary_bg_color text-header_color w-20 text-base box-border border-2 rounded-full cursor-pointer hover:bg-secondary_bg_color text-center">
-					<Link to="/signup">Sign up</Link>
-				</div> */}
-			{/* ) : ( */}
-
-			<div
-				className="text-xs
-							box-border 
-							border-2 
-							md:w-40
-							p-0.5
-							md:p-0
-							w-40
-							rounded-full 
-							cursor-pointer 
-							hover:bg-primary_bg_color 
-							mr-10 
-							text-center
-							md:text-base
-							ml-6
-							"
-			>
-				Sign Out
+		) : (
+			<div className="flex items-center gap-3 relative">
+				<Link to="/login">
+					<button className="border-2 border-solid	border-primary_bg_color rounded-full px-3 text-sm">
+						Log in
+					</button>
+				</Link>
+				<Link to="/signup">
+					<button className="font-bold text-header_color border-2 border-solid	border-primary_bg_color bg-primary_bg_color rounded-full px-3 text-sm">
+						Sign up
+					</button>
+				</Link>
 			</div>
+		);
+	};
 
-			<div className="">
-				<header className="bg-header_color h-10 flex items-center justify-center">
-					{isClosed ? (
-						<button
-							className="w-8 mb-4 md:w-12 md:mb-2"
-							aria-label="Open menu"
-							title="Open menu"
-							onClick={() => setClosed(false)}
-						>
-							<svg
-								aria-hidden="true"
-								fill="none"
-								strokeLinecap="round"
-								strokeLinejoin="round"
-								strokeWidth="2"
-								viewBox="0 0 24 24"
-								stroke="currentColor"
-							>
-								<path d="M4 6h16M4 12h16M4 18h16"></path>
-							</svg>
-						</button>
-					) : (
-						<button
-							className="w-8 mb-4 md:w-12 md:mb-2"
-							aria-label="Close menu"
-							title="Close menu"
-							onClick={() => setClosed(true)}
-						>
-							<svg
-								fill="none"
-								strokeLinecap="round"
-								strokeLinejoin="round"
-								strokeWidth="2"
-								viewBox="0 0 24 24"
-								stroke="currentColor"
-							>
-								<path d="M6 18L18 6M6 6l12 12"></path>
-							</svg>
-						</button>
-					)}
-				</header>
-			</div>
-			{/* )} */}
+	const closeClickingOutsideMenu = (e: any) => {
+		if (e.target.closest('.menu') === null) {
+			setOpen(false);
+		}
+	};
+
+	useEffect(() => {
+		document.addEventListener('click', closeClickingOutsideMenu);
+	}, []);
```

Comment:
I don't think we need to use document.addEventlistener() on React. Instead we can just use onClick on elements.
You can check [the official document](https://reactjs.org/docs/handling-events.html#:~:text=When%20using%20React%2C%20you%20generally,the%20element%20is%20initially%20rendered.&text=You%20have%20to%20be%20careful,are%20not%20bound%20by%20default.)
And with this code `closeClickingOutsideMenu()` always gets executed every time a user clicks on a page even though the menu is closed, and maybe this is a little problem.
To solve this, we can create an element as kind of overlay just under the dropdown menu. (image below).
But what do you think?
![Image from iOS (3)](https://user-images.githubusercontent.com/69213541/143506145-0279f917-017b-443d-8c15-a286dd6bf017.jpg)

Reply:
> When using React, you generally don’t need to call addEventListener to add listeners to a DOM element after it is created. Instead, just provide a listener when the element is initially rendered.

Probably you pointed out this one on the doc.  I intended to provide a listener when the first render occurs using `useEffect` with no independencies as this part says.

> And with this code `closeClickingOutsideMenu()` always gets executed every time a user clicks on a page even when the menu is closed

I agree with you. Could you add the overlay part for us, please?
Maybe the overlay has to move together with the menu.
We can talk about this later.

Thanks for reviewing thoroughly 🙏